File: backend/ExpenseTracker.Core/Entities/Expense.cs
namespace ExpenseTracker.Core.Entities
{
    public class Expense : BaseEntity
    {
        public string Title { get; set; } = string.Empty;
        public string? Description { get; set; }
        public decimal Amount { get; set; }
        public DateTime ExpenseDate { get; set; }
        public string? Notes { get; set; }

        // Foreign keys
        public int CategoryId { get; set; }
        public Guid UserId { get; set; }

        // Navigation properties
        public virtual Category Category { get; set; } = null!;
        public virtual User User { get; set; } = null!;
    }
}

File: frontend/src/types/expense.ts
export interface Expense {
  id: number;
  title: string;
  description?: string;
  amount: number;
  expenseDate: string;
  notes?: string;
  categoryId: number;
  userId: string;
  category: Category;
  createdAt: string;
  updatedAt?: string;
}

export interface Category {
  id: number;
  name: string;
  description?: string;
  categoryType: number;
}

export interface CreateExpenseDto {
  title: string;
  description?: string;
  amount: number;
  expenseDate: string;
  notes?: string;
  categoryId: number;
}

export interface UpdateExpenseDto {
  id: number;
  title: string;
  description?: string;
  amount: number;
  expenseDate: string;
  notes?: string;
  categoryId: number;
}

export interface ExpenseFilters {
  startDate?: string;
  endDate?: string;
  categoryId?: number;
  period?: 'week' | 'month' | 'year';
}